openupgrade_scripts/scripts/sale/16.0.1.2/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,93 @@
+---Models in module 'sale'---
+obsolete model report.all.channels.sales [sql_view]
+obsolete model report.sale.report_saleproforma [abstract]
+# NOTHING TO DO: obsolete model, does not exist in the database
+
+obsolete model sale.payment.acquirer.onboarding.wizard [transient]
+new model sale.payment.provider.onboarding.wizard [transient]
+# DONE: pre-migration: renamed model sale.payment.acquirer.onboarding.wizard to sale.payment.provider.onboarding.wizard
+
+---Fields in module 'sale'---
+sale         / account.analytic.applicability / business_domain (False)       : NEW selection_keys: ['bill', 'expense', 'general', 'invoice', 'purchase_order', 'sale_order'], mode: modify
+# NOTHING TO DO
+
+sale         / account.bank.statement.line / partner_shipping_id (many2one): module is now 'account' ('sale')
+sale         / account.move             / partner_shipping_id (many2one): module is now 'account' ('sale')
+sale         / account.move.line        / is_downpayment (boolean)      : NEW
+sale         / account.payment          / partner_shipping_id (many2one): module is now 'account' ('sale')
+# NOTHING TO DO
+
+sale         / payment.acquirer         / so_reference_type (selection) : DEL selection_keys: ['partner', 'so_name']
+sale         / payment.provider         / so_reference_type (selection) : NEW selection_keys: ['partner', 'so_name'], hasdefault: default
+# NOTHING TO DO
+
+sale         / sale.order               / show_update_fpos (boolean)    : NEW
+sale         / sale.order               / show_update_pricelist (boolean): not stored anymore
+# NOTHING TO DO: handle by ORM
+
+sale         / sale.order.line          / analytic_distribution_stored_char (char): NEW isfunction: function, stored
+# NOTHINGH TO DO: handle by ORM
```

Comment:
This the reason why we need the new analysis file because sometime it cause us some confusion
BTW, for the field `analytic_distribution_stored_char` which is in my lates analysis file it is `analytic_distribution`, i have created a dynamic method at https://github.com/OCA/OpenUpgrade/pull/4070/files#diff-90f733d1f5eba9ea514de973136064df20f321c35c019951653200d2c0e582d3R501-R515.
Please make a test with this situation then let me know it is working ok then i will decide to have it in the `openupgrade_160`